### PR TITLE
416 add determinism utility

### LINKED ISSUE #416: add determinism utility function
**Is your feature request related to a problem? Please describe.**
Add utility function to set determinism training easier.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #416 .

### Description
1. This PR implemented the determinism API to set the random seed for all modules.
2. And set seed for MONAI transforms before definition, so we don't need to set the random seed for transforms after definition anymore.
3. Also updated all the integration tests and notebooks that related to determinism training.

### Status
**Ready**

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### CODE REVIEW

File: monai/utils/misc.py [hunk 1]
```diff
@@ -99,3 +102,48 @@ def process_bar(index, count, bar_len=30, newline=False):
...
+
+def manual_seed(seed):
+    _seed = seed
+
+
+def get_seed():
+    return _seed
+
+
+def set_determinism(enable=True, seed=2147483647, additional_settings=None):
```
Comment (wyli) [inline monai/utils/misc.py:115]: I think the function name should be `set_seed`, and it enables deterministic behaviour when `seed is not None`. so the expected behaviours would be:
- user runs `set_seed(seed=42)`, enables deterministic modules for all
- user does nothing, or runs `set_seed(seed=None)`, run all modules without deterministic modules

File: monai/utils/misc.py [hunk 2]
```diff
@@ -99,3 +102,48 @@ def process_bar(index, count, bar_len=30, newline=False):
     print(f"{index}/{count} {bar:s}  ", end=end)
     if index == count:
         print("")
+
+
+def manual_seed(seed):
+    _seed = seed
```
Comment (wyli) [inline monai/utils/misc.py:108]: I think it should be:
```
global _seed
_seed = seed
```
may be:
```
global _seed
_seed = int(seed)
```
(could we have a unit test for this?)

**Comment by Nic-Ma** [comment]
> This looks like a very nice end-user simplification!

thanks for your review here!

File: monai/utils/misc.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/utils/misc.py:115]: updated in the latest patch.
Thanks.

File: monai/utils/misc.py (hunk 2 above)
Comment (Nic-Ma) [reply monai/utils/misc.py:108]: updated in the latest patch.
Thanks.

**Pull request merged**